src/Parallel/PhaseControl/PhaseControlTags.hpp

```diff
@@ -84,4 +91,78 @@ using reduction_data = Parallel::ReductionData<Parallel::ReductionDatum<
     tuples::tagged_tuple_from_typelist<TagsAndCombinesPresent>,
     TaggedTupleCombine>>;
 
+namespace OptionTags {
+/// Option tag for the collection of triggers that indicate synchronization
+/// points at which phase changes should be considered, and the associated
+/// `PhaseChange` objects for making the phase change decisions.
+///
+/// \note The nested collection types for this option tag gives the yaml
+/// format the slightly unusual form:
+///
+/// ```
+/// PhaseChangeAndTriggers:
+///   - - Trigger1
+///     - - PhaseChange1
+///       - PhaseChange2
+///     - Trigger2
+///     - - PhaseChange3
+///       - PhaseChange4
+/// ```
+template <typename PhaseChangeRegistrars, typename TriggerRegistrars>
+struct PhaseChangeAndTriggers : db::SimpleTag {
+  using phase_control_type = PhaseChange<PhaseChangeRegistrars>;
+  using trigger_type = Trigger<TriggerRegistrars>;
+  static constexpr Options::String help{
+    "A map from triggers to collections of phase change objects to determine "
+    "runtime phase control-flow decisions"};
+
+  using type =
+      std::vector<std::pair<std::unique_ptr<trigger_type>,
+                            std::vector<std::unique_ptr<phase_control_type>>>>;
+};
+}  // namespace OptionTags
+
+namespace Tags {
+/// Tag for the collection of triggers that indicate synchronization points at
+/// which phase changes should be considered, and the associated `PhaseChange`
+/// objects for making the phase change decisions.
+template <typename PhaseChangeRegistrars, typename TriggerRegistrars>
+struct PhaseChangeAndTriggers : db::SimpleTag {
+  using phase_control_type = PhaseChange<PhaseChangeRegistrars>;
+  using trigger_type = Trigger<TriggerRegistrars>;
+  using type =
+      std::vector<std::pair<std::unique_ptr<trigger_type>,
+                            std::vector<std::unique_ptr<phase_control_type>>>>;
+
+  using option_tags =
+      tmpl::list<OptionTags::PhaseChangeAndTriggers<PhaseChangeRegistrars,
+                                                    TriggerRegistrars>>;
+  static constexpr bool pass_metavariables = false;
+  static type create_from_options(
+      const type& phase_control_and_triggers) noexcept {
+    return deserialize<type>(
+        serialize<type>(phase_control_and_triggers).data());
+  }
+};
+
+struct UsePhaseChangeArbitration {
```

Comment:
what is this? please document.  and why is it in the `Tags` namespace (perhaps it needs its own namespace 🤷 )